File: pset6/readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

#Initiate words, letters & sentences to count
words = 1
letters = 0
sentences = 0

#Count letter & sentence
for i in range(len(text)):
    if (text[i].isspace() == True):
        words += 1

    if (text[i].isalpha() == True):
        letters += 1

    if (text[i] == '.' or text[i] == '?' or text[i] == '!'):
        sentences += 1

#Average
L = letters / words * 100
S = sentences / words * 100

#Plug in Coleman-Liau formula to get grade level
X = round(0.0588 * L - 0.296 * S - 15.8)

#Print out grade level
if X < 16 and X >= 1:
    print(f"Grade {X}")

else:
    if X >= 16:
        print("Grade 16+")

    if X < 1:
        print("Before Grade 1")

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>


int main(void)
{
    //Get text
    string text = get_string("Text: ");
    float letters = 0;
    float words = 1;
    float sentences = 0;
    
    //Count letters
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        if (isspace(text[i]))
        {
            words++;
        }
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }
    //Calculate average
    float L = letters / words * 100;
    float S = sentences / words * 100;
    
    //Readability
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int result = round(index);
    
    //Output
    if (result >= 1 && result <= 16)
    {
            printf("Grade %i\n", result);
    }
    else
    {
        if (result < 1)
        {
            printf("Before Grade 1\n");
        }
        if (result > 16)
        {
            printf("Grade 16+\n");
        }
    }
}